models/login_source.go

```diff
@@ -403,6 +404,20 @@ func LoginViaLDAP(user *User, login, password string, source *LoginSource, autoR
 
 	var isAttributeSSHPublicKeySet = len(strings.TrimSpace(source.LDAP().AttributeSSHPublicKey)) > 0
 
+	// Update User admin flag if exist
+	if isExist, err := IsUserExist(0, sr.Username); err != nil {
+		return nil, err
+	} else if isExist {
+		if !user.ProhibitLogin && len(source.LDAP().AdminFilter) > 0 && user.IsAdmin != sr.IsAdmin {
+			// Change existing admin flag only if AdminFilter option is set
+			err = UpdateUserCols(user, "is_admin")
```

Comment:
should there be `user.IsAdmin = sr.IsAdmin` before calling `UpdateUserCols`?

Reply:
:D done @lafriks